### PR TITLE
Add a new setting to enable the invoice date column in the orders list

### PR DESCRIPTION
**Pull request by YordanSoares**
Closes: #308

### CODE REVIEW

File: includes/class-wcpdf-admin.php [hunk 1]
```diff
@@ -257,6 +280,34 @@ public function invoice_number_column_data( $column ) {
...
+	 */
+	public function invoice_date_column_data( $column ) {
+		global $post, $the_order;
+
+		if ( $column == 'pdf_invoice_date' ) {
+			$this->disable_storing_document_settings();
+			if ( empty( $the_order ) || WCX_Order::get_id( $the_order ) != $post->ID ) {
+				$order = WCX::get_order( $post->ID );
+				if ( $invoice = wcpdf_get_invoice( $order ) ) {
+					$output_format = get_option( 'date_format' ) . ' '.get_option( 'time_format' );
```
Comment (alexmigf) [inline includes/class-wcpdf-admin.php:295]: Are you still going to use this?

File: includes/class-wcpdf-admin.php [hunk 2]
```diff
@@ -737,12 +795,16 @@ public function pre_get_posts_sort_by_invoice_number( $query ) {
 			$query->set( 'meta_key', '_wcpdf_invoice_number' );
 			$query->set( 'orderby', apply_filters( 'wpo_wcpdf_invoice_number_column_orderby', 'meta_value' ) );
 		}
+		if( 'pdf_invoice_date' == $orderby ) {
```
Comment (alexmigf) [inline includes/class-wcpdf-admin.php:798]: `elseif` here? Or maybe a `switch`.

File: includes/class-wcpdf-admin.php [hunk 3]
```diff
@@ -724,11 +775,18 @@ public function invoice_number_column_sortable( $columns ) {
...
+	public function invoice_date_column_sortable( $columns ) {
+		$columns['pdf_invoice_date'] = 'pdf_invoice_date';
+		return $columns;
+	}
 
 	/**
 	 * Pre WC3.X sorting
 	 */
-	public function pre_get_posts_sort_by_invoice_number( $query ) {
+	public function pre_get_posts_sort_by_invoice_number_and_invoice_date( $query ) {
```
Comment (alexmigf) [inline includes/class-wcpdf-admin.php:789]: Not sure about this function name, maybe would be better to have a general one to allow adding more columns in the future?

eg. `pre_get_posts_sort_by_column`

**Review by alexmigf — CHANGES_REQUESTED** [review]

File: includes/class-wcpdf-admin.php [hunk 4]
```diff
@@ -215,46 +215,129 @@ public function add_listing_actions( $order ) {
...
+			case 'invoice_number_column':
+				echo $invoice->get_number();
 				do_action( 'wcpdf_invoice_number_column_end', $order );
-			} else {
-				if ( $invoice = wcpdf_get_invoice( $the_order ) ) {
-					echo $invoice->get_number();
+				break;
+			case 'invoice_date_column':
+				if ( ! empty( $date = $invoice->get_date() ) ) {
+					echo $date->date_i18n( wcpdf_date_format() );
```
Comment (YordanSoares) [inline includes/class-wcpdf-admin.php:275]: @alexmigf can this be filtered to display the time too?

Comment (alexmigf) [reply includes/class-wcpdf-admin.php:275]: Yes https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/e354f5cce95ea2feb28f1a84cbddb25a8176c886/includes/wcpdf-functions.php#L249

Comment (YordanSoares) [reply includes/class-wcpdf-admin.php:275]: But it can be limited to this context somehow?

Comment (alexmigf) [reply includes/class-wcpdf-admin.php]: Done here [93d183f](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/309/commits/93d183f4eb8f5b038d95845d9a6c41b35d5b96c9)

**Pull request merged**